Tracking of projects that fail to run coverage job

- [x] **bad_example**
Failed to get list of targets from "https://storage.googleapis.com/clusterfuzz-builds/bad_example/targets.list.address".
as the target is not being run on CF
- [x] **chakra**
The same root cause: Failed to get list of targets from "https://storage.googleapis.com/clusterfuzz-builds/chakra/targets.list.address".
- [x] **dlplibs**
the disk is full, cannot unpack the corpus:
```
I  Step #4: /corpus/mswksfuzzer/5baeffd6604290d2cbea9a2112181f6c7b91ea0c:  write error (disk full?).  Continue? (y/n/^C)  
I  Step #4: warning:  /corpus/mswksfuzzer/5baeffd6604290d2cbea9a2112181f6c7b91ea0c is probably truncated 
I  Step #4: checkdir:  cannot create extraction directory: /corpus/mswrdfuzzer 
I  Step #4:            No space left on device 
I  Step #4: checkdir:  cannot create extraction directory: /corpus/multiplanfuzzer 
I  Step #4:            No space left on device 
```
- [x] **example**
download_corpus failed, target is disabled on CF
- [x] **firefox**
the disk is full, cannot finish compilation
```
I  Step #2: + find media/webrtc/trunk/webrtc/test/fuzzers/corpora/stun-corpus -type f -exec zip -qju /workspace/out/profile/StunParser_seed_corpus.zip '{}' ';' 
I  Step #2: + cp media/webrtc/trunk/webrtc/test/fuzzers/corpora/stun.tokens /workspace/out/profile/StunParser.dict 
I  Step #2: cp: error writing '/workspace/out/profile/work/obj-fuzz/_tests/testing/mochitest/chrome/toolkit/mozapps/update/tests/data/updater': No space left on device 
I  Step #2: cp: cannot create directory '/workspace/out/profile/work/obj-fuzz/_tests/testing/mochitest/chrome/toolkit/components': No space left on device 
``` 
- [x] **fuzzing-puzzles**
The project is disabled: Failed to get list of targets from "https://storage.googleapis.com/clusterfuzz-builds/fuzzing-puzzles/targets.list.address".
- [x] **glib**
Compilation failure. Weird, as I've added `-ldl` flag. Looks like the project does linking using `clang`, not `clang++`:
```
I  Step #2: [773/955] Linking target gio/gsettings. 
I  Step #2: FAILED: gio/gsettings  
I  Step #2: clang  -o gio/gsettings 'gio/gio@@gsettings@exe/gsettings-tool.c.o' -Wl,--no-undefined -Wl,--as-needed -O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fprofile-instr-generate -fcoverage-mapping -pthread -Wl,-ldl -Wno-unused-command-line-argument -Wl,--start-group gio/libgio-2.0.a gio/xdgmime/libxdgmime.a gio/inotify/libinotify.a glib/libglib-2.0.a glib/libcharset/libcharset.a glib/pcre/libpcre.a gobject/libgobject-2.0.a subprojects/libffi/src/libffi.a subprojects/zlib-1.2.11/libz.a gmodule/libgmodule-2.0.a -lresolv -Wl,--end-group   
I  Step #2: gmodule/libgmodule-2.0.a(gmodule.c.o): In function `_g_module_self': 
I  Step #2: /work/meson/../../src/glib/gmodule/gmodule-dl.c:125: undefined reference to `dlopen' 
I  Step #2: gmodule/libgmodule-2.0.a(gmodule.c.o): In function `_g_module_open': 
I  Step #2: /work/meson/../../src/glib/gmodule/gmodule-dl.c:98: undefined reference to `dlopen' 
I  Step #2: gmodule/libgmodule-2.0.a(gmodule.c.o): In function `_g_module_close': 
I  Step #2: /work/meson/../../src/glib/gmodule/gmodule-dl.c:150: undefined reference to `dlclose' 
I  Step #2: gmodule/libgmodule-2.0.a(gmodule.c.o): In function `_g_module_symbol': 
I  Step #2: /work/meson/../../src/glib/gmodule/gmodule-dl.c:163: undefined reference to `dlsym' 
I  Step #2: gmodule/libgmodule-2.0.a(gmodule.c.o): In function `fetch_dlerror': 
I  Step #2: /work/meson/../../src/glib/gmodule/gmodule-dl.c:80: undefined reference to `dlerror' 
I  Step #2: clang-7: error: linker command failed with exit code 1 (use -v to see invocation) 
```
- [x] **jsc**
The project is disabled: Failed to get list of targets from "https://storage.googleapis.com/clusterfuzz-builds/jsc/targets.list.address".
- [x] **libchewing**
The target doesn't have corpus backups as it's disabled: https://github.com/google/oss-fuzz/blob/master/projects/libchewing/project.yaml#L2
- [x] **libprotobuf-mutator**
Compilation failure without a reasonable error message:
```
I  Step #2: [208/209] Building CXX object CMakeFiles/protoc.dir/src/libprotobuf-mutator/build/external.protobuf/src/external.protobuf/src/google/protobuf/compiler/main.cc.o 
I  Step #2: [209/209] Linking CXX executable protoc 
I  Step #2: ninja: build stopped: subcommand failed. 
I  Finished Step #2 
I  ERROR 
I  ERROR: build step 2 "gcr.io/oss-fuzz/libprotobuf-mutator" failed: exit status 1 
```
Regular build has the same problem though: https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=10003
- [x] **libpsl**
Compilation failure:
```
I  Step #2:   CCLD     psl 
I  Step #2: /usr/bin/ld: /src/deps/lib/libicuuc.a(putil.ao): undefined reference to symbol 'floor@@GLIBC_2.2.5' 
I  Step #2: //lib/x86_64-linux-gnu/libm.so.6: error adding symbols: DSO missing from command line 
I  Step #2: clang-7: error: linker command failed with exit code 1 (use -v to see invocation) 
I  Step #2: make[2]: *** [psl] Error 1 
I  Step #2: Makefile:468: recipe for target 'psl' failed 
```
- [x] **libssh**
Compilation failure, the same as the regular build has.
```
I  Step #2: CMake Error at cmake/Modules/CheckCCompilerFlagSSP.cmake:19 (include_guard): 
I  Step #2:   Unknown CMake command "include_guard". 
I  Step #2: Call Stack (most recent call first): 
I  Step #2:   CompilerChecks.cmake:2 (include) 
I  Step #2:   CMakeLists.txt:31 (include) 
I  Step #2:  
I  Step #2:  
I  Step #2: -- Configuring incomplete, errors occurred! 
I  Step #2: See also "/src/libssh/build/CMakeFiles/CMakeOutput.log". 
```
- [x] **mercurial**
Corpus download failed, probably because https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=9694
xdiff_fuzzer seems to be crazy slow so corpus pruning never finishes since early July
- [x] **nss**
No fuzz targets (weird): Failed to get list of targets from "https://storage.googleapis.com/clusterfuzz-builds/nss/targets.list.address".
- [x] **openvswitch**
Build failure:
```
I  Step #2: + clang++ -O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fprofile-instr-generate -fcoverage-mapping -pthread -Wl,-ldl -Wno-unused-command-line-argument -stdlib=libc++ /workspace/out/profile/flow_extract_target.o ./lib/.libs/libopenvswitch.a -lz -lssl -lcrypto -latomic -lFuzzingEngine -o /workspace/out/profile/flow_extract_fuzzer 
I  Step #2: ./lib/.libs/libopenvswitch.a(async-append-aio.o): In function `async_append_write': 
I  Step #2: /src/openvswitch/lib/async-append-aio.c:144: undefined reference to `aio_write' 
I  Step #2: ./lib/.libs/libopenvswitch.a(async-append-aio.o): In function `async_append_wait': 
I  Step #2: /src/openvswitch/lib/async-append-aio.c:99: undefined reference to `aio_error' 
I  Step #2: /src/openvswitch/lib/async-append-aio.c:106: undefined reference to `aio_suspend' 
I  Step #2: /src/openvswitch/lib/async-append-aio.c:108: undefined reference to `aio_return' 
I  Step #2: clang-7: error: linker command failed with exit code 1 (use -v to see invocation) 

```
- [x] **postgis**
Corpus download failed. Missing corpus backup for postgis_wkb_import_fuzzer.
- [x] **proj4**
Build failure (messing up with `atexit` which is used for dumping coverage):
```
I  Step #2: clang-7: error: linker command failed with exit code 1 (use -v to see invocation) 
I  Step #2: /usr/bin/ld: .libs/basic_test: hidden symbol `atexit' in /usr/lib/x86_64-linux-gnu/libc_nonshared.a(atexit.oS) is referenced by DSO 
I  Step #2: /usr/bin/ld: final link failed: Bad value 
I  Step #2: make[2]: *** [proj_errno_string_test] Error 1 
I  Step #2: Makefile:424: recipe for target 'proj_errno_string_test' failed 
I  Step #2: clang-7: error: linker command failed with exit code 1 (use -v to see invocation) 
I  Step #2: make[2]: *** [basic_test] Error 1 
I  Step #2: Makefile:416: recipe for target 'basic_test' failed 
I  Step #2: Makefile:369: recipe for target 'all-recursive' failed 
I  Step #2: make[1]: *** [all-recursive] Error 1 
I  Step #2: make: *** [all-recursive] Error 1 
I  Step #2: Makefile:448: recipe for target 'all-recursive' failed 

```
- [x] **spidermonkey**
No fuzz targets, WAI: Failed to get list of targets from "https://storage.googleapis.com/clusterfuzz-builds/spidermonkey/targets.list.address".
- [x] **strongswan**
Build failure. Not sure what's going on, by adding `-lm` likely will solve the problem:

```
I  Step #2: rm -f libFuzzerLocal.a 
I  Step #2: ar cru libFuzzerLocal.a libFuzzerLocal.o ../src/libstrongswan/libstrongswan.la 
I  Step #2: ar: `u' modifier ignored since `D' is the default (see `U') 
I  Step #2: ranlib libFuzzerLocal.a 
I  Step #2: /usr/lib/libFuzzingEngine.a(FuzzerLoop.o): In function `ceil': 
I  Step #2: /usr/local/bin/../include/c++/v1/math.h:834: undefined reference to `ceilf' 
I  Step #2: /usr/local/bin/../include/c++/v1/math.h:834: undefined reference to `ceilf' 
```

--- a/infra/gcb/build_and_run_coverage.py
+++ b/infra/gcb/build_and_run_coverage.py
@@ -46,6 +46,8 @@ def get_build_steps(project_dir):
 
     # Exit with 0 not to mark code coverage job as failed in a case when project
     # did not have any successful builds and there are no fuzz targets recorded.
+    # The script should print build_id, print a special message instead.
+    print 'NO_FUZZ_TARGETS_FOUND'
     exit(0)
 
   project_yaml = build_project.load_project_yaml(project_dir)

--- a/infra/gcb/build_and_run_coverage.py
+++ b/infra/gcb/build_and_run_coverage.py
@@ -46,8 +46,8 @@ def get_build_steps(project_dir):
 
     # Exit with 0 not to mark code coverage job as failed in a case when project
     # did not have any successful builds and there are no fuzz targets recorded.
-    # The script should print build_id, print a special message instead.
-    print 'NO_FUZZ_TARGETS_FOUND'
+    # The script should print build_id, print '0' as a special value.
+    print '0'
     exit(0)
 
   project_yaml = build_project.load_project_yaml(project_dir)

--- a/infra/gcb/jenkins_config/coverage_job.xml
+++ b/infra/gcb/jenkins_config/coverage_job.xml
@@ -47,8 +47,8 @@ cd $WORKSPACE/oss-fuzz/infra/gcb
 pip install -r requirements.txt
 project_dir=$WORKSPACE/oss-fuzz/projects/$(basename $JOB_NAME)
 build_id=$(python build_and_run_coverage.py $project_dir)
-if [[ "$build_id" == "NO_FUZZ_TARGETS_FOUND" ]]; then
-  echo "No fuzz targets found for the project. Skipping code coverage job."
+if [[ "$build_id" == "0" ]]; then
+  echo "Intentionally skipping code coverage job."
 else
   python wait_for_build.py $build_id
 fi

--- a/infra/gcb/jenkins_config/coverage_job.xml
+++ b/infra/gcb/jenkins_config/coverage_job.xml
@@ -47,7 +47,11 @@ cd $WORKSPACE/oss-fuzz/infra/gcb
 pip install -r requirements.txt
 project_dir=$WORKSPACE/oss-fuzz/projects/$(basename $JOB_NAME)
 build_id=$(python build_and_run_coverage.py $project_dir)
-python wait_for_build.py $build_id
+if [[ "$build_id" == "NO_FUZZ_TARGETS_FOUND" ]]; then
+  echo "No fuzz targets found for the project. Skipping code coverage job."
+else
+  python wait_for_build.py $build_id
+fi
 </command>
     </hudson.tasks.Shell>
   </builders>

